GAE/src/org/akvo/flow/api/app/DataPointServlet.java

```diff
@@ -77,63 +80,89 @@ protected RestRequest convertRequest() throws Exception {
     protected RestResponse handleRequest(RestRequest req) throws Exception {
         DataPointRequest dpReq = (DataPointRequest) req;
         RestResponse res = new RestResponse();
-        if (dpReq.getSurveyId() != null) {
-            //Find the device (if any)
-            DeviceDAO deviceDao = new DeviceDAO();
-            Device device = deviceDao.getDevice(dpReq.getAndroidId(), null, null);
-            if (device != null) {
-                log.info("Found device id: " + device.getKey().getId());
-                log.fine("Found device: " + device);
-                List<SurveyedLocale> dpList = getDataPointList(dpReq.getSurveyId(), device.getKey().getId());
-                if (dpList == null) {
-                    res.setCode(String.valueOf(HttpServletResponse.SC_NOT_FOUND));
-                    res.setMessage("No assignment was found");
-                    return res;
-                }
-                res = convertToResponse(dpList, dpReq.getSurveyId());
-                return res;
-            }
-            res.setCode(String.valueOf(HttpServletResponse.SC_NOT_FOUND));
-            res.setMessage("Unknown device");
-        } else {
+        if (dpReq.getSurveyId() == null) {
             res.setCode(String.valueOf(HttpServletResponse.SC_FORBIDDEN));
```

Comment:
We're changing the status code from `HTTP 404` to `HTTP 403`. If we don't find the *device* or the *datapoint list* is empty, we keep using `HTTP 404`. Why this case is different?

Reply:
The `403` was always there in the `else` branch so I have just made it more explicit. I am not sure how the older versions process the returned status so I did not want to change it. Otherwise I actually wanted to return a `401` if there is no `surveyId`